<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Example data in the sceptredata package</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Example data in the
<code>sceptredata</code> package</h1>



<p>The <code>sceptredata</code> package contains three example
single-cell CRISPR screen datasets for use in the examples of the
<code>sceptre</code> and <code>ondisc</code> packages. We describe each
of the example datasets here.</p>
<div id="high-multiplicity-of-infection-crispri-screen-of-candidate-enhancers" class="section level2">
<h2>High-multiplicity-of-infection CRISPRi screen of candidate
enhancers</h2>
<p>The first example dataset contained within <code>sceptredata</code>
is a high-multiplicity-of-infection (MOI) CRISPRi screen of candidate
enhancers in K562 cells <a href="https://pubmed.ncbi.nlm.nih.gov/30612741/">(Gasperini, 2019)</a>.
The original authors aimed to link candidate enhancers to genes by
testing for changes in gene expression in response to CRISPR
perturbations of the candidate enhancers. The authors designed a library
of gRNAs to target 5,779 candidate enhancers and 381 transcription start
sites (TSSs). Each genomic element was targeted by two gRNAs (with a
subset of the candidate enhancers targeted by four gRNAs). Additionally,
the authors designed 100 negative control gRNAs, each of which targeted
a gene desert or no location in the genome at all. The authors sequenced
207,324 single cells, measuring the gene and gRNA expression profile of
each.</p>
<p>The <code>sceptredata</code> package contains a downsampled version
of the Gasperini data, which is stored within the
<code>highmoi_example_data</code> list. Evaluating
<code>data(highmoi_example_data)</code> in the console loads the example
high-MOI CRISPRi data into the global namespace.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(sceptredata)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">data</span>(highmoi_example_data)</span></code></pre></div>
<p>The data consist of several pieces. First, <code>grna_matrix</code>
is the gRNA-by-cell expression matrix. gRNAs and cells alike are
downsampled: the matrix contains 95 gRNAs and 45,919 cells. The gRNAs
are downsampled such that there are 50 candidate-enhancer-targeting
gRNAs (targeting 25 distinct candidate enhancers), 20 positive control
gRNAs (targeting 10 TSSs), and 25 negative control gRNAs. The cells are
downsampled such that each contains at least one gRNA.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>grna_matrix <span class="ot">&lt;-</span> highmoi_example_data<span class="sc">$</span>grna_matrix</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="fu">dim</span>(grna_matrix)</span></code></pre></div>
<pre><code>## [1]    95 45919</code></pre>
<p>Second, <code>response_matrix</code> is the gene-by-cell expression
matrix. This matrix is also downsampled, containing 526 genes and the
same 45,919 cells as the gRNA matrix. The target gene of each
TSS-targeting gRNA is included in the matrix. Moreover, the matrix
contains about 500 genes in close physical proximity to the target
locations of the (downsampled) candidate-enhancer-targeting gRNAs.
Finally, two randomly-selected mitochondrial genes are included in the
matrix.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>response_matrix <span class="ot">&lt;-</span> highmoi_example_data<span class="sc">$</span>response_matrix</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="fu">dim</span>(response_matrix)</span></code></pre></div>
<pre><code>## [1]   526 45919</code></pre>
<p>Third, <code>extra_covariates</code> is a data frame containing one
column, namely <code>batch</code>, which indicates the batch
(<code>b1</code> or <code>b2</code>) in which each cell was
sequenced.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>extra_covariates <span class="ot">&lt;-</span> highmoi_example_data<span class="sc">$</span>extra_covariates</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="fu">head</span>(extra_covariates)</span></code></pre></div>
<pre><code>##   batch
## 1    b1
## 2    b1
## 3    b1
## 4    b1
## 5    b1
## 6    b1</code></pre>
<p>Fourth, <code>gene_names</code> is a vector giving the human-readable
name of each gene. Genes prepended with the string “MT-” or “mt-” are
taken as mitochondrial genes and used to compute the
<code>response_p_mito</code> covariate in <code>sceptre</code>.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>gene_names <span class="ot">&lt;-</span> highmoi_example_data<span class="sc">$</span>gene_names</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="fu">head</span>(gene_names)</span></code></pre></div>
<pre><code>## [1] &quot;NUCKS1&quot;  &quot;RBBP5&quot;   &quot;CDK18&quot;   &quot;RAB29&quot;   &quot;DSTYK&quot;   &quot;SLC41A1&quot;</code></pre>
<p>Finally, <code>grna_target_data_frame_highmoi</code> is a data frame
containing information about the target of each gRNA.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="fu">data</span>(grna_target_data_frame_highmoi)</span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a><span class="fu">head</span>(grna_target_data_frame_highmoi)</span></code></pre></div>
<pre><code>##        grna_id     grna_target   chr    start      end
## 1 grna_CCGGGCG ENSG00000069482 chr11 68451943 68451958
## 2 grna_TGGCGGC ENSG00000069482 chr11 68451958 68451974
## 3 grna_AAGGCCG ENSG00000100316 chr22 39715775 39715790
## 4 grna_GACGCCG ENSG00000100316 chr22 39715790 39715806
## 5 grna_CACACCC ENSG00000104131 chr15 44829255 44829270
## 6 grna_GCTCACA ENSG00000104131 chr15 44829270 44829286</code></pre>
<p><code>grna_target_data_frame_highmoi</code> contains the columns
<code>grna_id</code> (the ID of an individual gRNA),
<code>grna_target</code> (the genomic target of the given gRNA), and
<code>chr</code>, <code>start</code>, and <code>end</code> (the
chromosome, start coordinate, and end coordinate, respectively, of the
target location of the given gRNA). Non-targeting gRNAs are assigned a
<code>grna_target</code> of <code>&quot;non-targeting&quot;</code>.</p>
<p>The <code>sceptredata</code> package also stores the example high-MOI
CRISPRi data in 10X Cell Ranger format. The data are contained in two
directories (namely, <code>gem_group_1</code> and
<code>gem_group_2</code>), which correspond to cells sequenced in
different batches. One can determine the location of these directories
on her computer as follows.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a>directories <span class="ot">&lt;-</span> <span class="fu">paste0</span>(</span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a>  <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="at">package =</span> <span class="st">&quot;sceptredata&quot;</span>), </span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a>  <span class="st">&quot;/highmoi_example/gem_group_&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span></span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a>)</span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a>directories <span class="co"># file paths to the example data on your computer</span></span></code></pre></div>
<pre><code>## [1] &quot;/private/var/folders/7v/5sqjgh8j28lgf8qx3gbtq1h00000gp/T/RtmpNaqVEz/temp_libpath13ccc6abdb76e/sceptredata/extdata/highmoi_example/gem_group_1&quot;
## [2] &quot;/private/var/folders/7v/5sqjgh8j28lgf8qx3gbtq1h00000gp/T/RtmpNaqVEz/temp_libpath13ccc6abdb76e/sceptredata/extdata/highmoi_example/gem_group_2&quot;</code></pre>
<p>Each of these directories contains three files:
<code>barcodes.tsv.gz</code>, <code>features.tsv.gz</code>, and
<code>matrix.mtx.gz</code>.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="fu">list.files</span>(directories[<span class="dv">1</span>])</span></code></pre></div>
<pre><code>## [1] &quot;barcodes.tsv.gz&quot; &quot;features.tsv.gz&quot; &quot;matrix.mtx.gz&quot;</code></pre>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a><span class="fu">list.files</span>(directories[<span class="dv">2</span>])</span></code></pre></div>
<pre><code>## [1] &quot;barcodes.tsv.gz&quot; &quot;features.tsv.gz&quot; &quot;matrix.mtx.gz&quot;</code></pre>
<p>The expression data are stored in “feature barcode matrix” format.
Users can learn more about this file format by consulting the 10X
Genomics <a href="https://www.10xgenomics.com/support/software/cell-ranger/latest/analysis/outputs/cr-outputs-mex-matrices">documentation</a>.</p>
</div>
<div id="low-moi-crisprko-screen-of-gene-transcription-start-sites" class="section level2">
<h2>Low-MOI CRISPRko screen of gene transcription start sites</h2>
<p>The second example dataset contained within <code>sceptredata</code>
is a low-MOI CRISPRko screen of gene TSSs in THP1 cells <a href="https://pubmed.ncbi.nlm.nih.gov/33649593/">(Papalexi, 2021)</a>.
The original authors aimed to link perturbations of genes to changes in
the expression of <em>other</em> genes. The authors designed 101
targeting gRNAs to target 26 gene TSSs (with four gRNAs per TSS, on
average). The authors additionally designed nine non-targeting gRNAs.
The authors profiled the gene and gRNA expression profiles of 20,729
single cells. (The authors additionally measured the expression level of
four cell-surface proteins; the protein expressions are not included as
part of the example data.) The low-MOI CRISPRko data are stored in the
<code>lowmoi_example_data</code> list; we can load this list into the
global namespace via a call to <code>data()</code>.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a><span class="fu">data</span>(lowmoi_example_data)</span></code></pre></div>
<p>The data consist of four pieces. First, <code>grna_matrix</code> is
the gRNA-by-cell matrix of gRNA UMI counts. <code>grna_matrix</code> is
unaltered; it contains all 110 gRNAs and 20,729 cells.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a>grna_matrix <span class="ot">&lt;-</span> lowmoi_example_data<span class="sc">$</span>grna_matrix</span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a><span class="fu">dim</span>(grna_matrix)</span></code></pre></div>
<pre><code>## [1]   110 20729</code></pre>
<p>Next, <code>response_matrix</code> is the gene-by-cell expression
matrix. The gene expression matrix is downsampled to contain 299 genes.
Nine of these genes (namely, <em>STAT1</em>, <em>JAK2</em>,
<em>CMTM6</em>, <em>STAT2</em>, <em>UBE2L6</em>, <em>STAT3</em>,
<em>TNFRSF14</em>, <em>IFNGR2</em>, <em>NFKBIA</em>) are among the genes
targeted by the targeting gRNAs. Next, one of the genes is a
randomly-selected mitochondrial gene. The remaining genes are randomly
sampled from the set of remaining genes.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a>response_matrix <span class="ot">&lt;-</span> lowmoi_example_data<span class="sc">$</span>response_matrix</span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a><span class="fu">dim</span>(response_matrix)</span></code></pre></div>
<pre><code>## [1]   299 20729</code></pre>
<p>Additionally, <code>extra_covariates</code> is a data frame
containing a single column — <code>bio_rep</code> — indicating the
biological replicate in which a given cell was sequenced
(<code>rep_1</code>, <code>rep_2</code>, or <code>rep_3</code>).</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a>extra_covariates <span class="ot">&lt;-</span> lowmoi_example_data<span class="sc">$</span>extra_covariates</span>
<span id="cb23-2"><a href="#cb23-2" tabindex="-1"></a><span class="fu">head</span>(extra_covariates)</span></code></pre></div>
<pre><code>##   bio_rep
## 1   rep_1
## 2   rep_1
## 3   rep_1
## 4   rep_1
## 5   rep_1
## 6   rep_1</code></pre>
<p>Finally, <code>grna_target_data_frame</code> is a data frame mapping
each individual gRNA (<code>grna_id</code>) to the gene that it targets
(<code>grna_target</code>). Again, non-targeting gRNAs are assigned a
<code>grna_target</code> string of <code>&quot;non-targeting&quot;</code>.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a>grna_target_data_frame <span class="ot">&lt;-</span> lowmoi_example_data<span class="sc">$</span>grna_target_data_frame</span>
<span id="cb25-2"><a href="#cb25-2" tabindex="-1"></a><span class="fu">head</span>(grna_target_data_frame)</span></code></pre></div>
<pre><code>##   grna_id grna_target
## 1  ATF2g1        ATF2
## 2  ATF2g2        ATF2
## 3  ATF2g3        ATF2
## 4  ATF2g4        ATF2
## 5  BRD4g1        BRD4
## 6  BRD4g2        BRD4</code></pre>
</div>
<div id="parse-data" class="section level2">
<h2>Parse data</h2>
<p>The third dataset is an example single-cell CRISPR screen of LN18
cells produced by <a href="https://www.parsebiosciences.com">Parse
Biosciences</a>. The example dataset included in the
<code>sceptredata</code> package is a subset of the <a href="https://www.parsebiosciences.com/datasets/technical-performance-of-crispr-detect-in-cell-lines/">sample
single-cell CRISPR screen dataset</a> available on the Parse website.
The sample dataset containins 50,470 cells, 58,395 genes, and three
gRNAs. Unfortunately, the genomic target of the gRNAs in the dataset is
not indicated. The example Parse data are stored in the following
machine-specific directory.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a>directory <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="at">package =</span> <span class="st">&quot;sceptredata&quot;</span>), <span class="st">&quot;/parse_example/&quot;</span>)</span>
<span id="cb27-2"><a href="#cb27-2" tabindex="-1"></a>directory <span class="co"># location of the data on your computer</span></span></code></pre></div>
<pre><code>## [1] &quot;/private/var/folders/7v/5sqjgh8j28lgf8qx3gbtq1h00000gp/T/RtmpNaqVEz/temp_libpath13ccc6abdb76e/sceptredata/extdata/parse_example/&quot;</code></pre>
<p>This directory contains four files: <code>all_grnas.csv</code>,
<code>grna_mat.mtx</code>, <code>all_genes.csv</code>, and
<code>gene_mat.mtx</code>.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" tabindex="-1"></a><span class="fu">list.files</span>(directory)</span></code></pre></div>
<pre><code>## [1] &quot;all_genes.csv&quot; &quot;all_grnas.csv&quot; &quot;gene_mat.mtx&quot;  &quot;grna_mat.mtx&quot;</code></pre>
<p><code>grna_mat.mtx</code> is a matrix market (<code>.mtx</code>) file
containing the gRNA expression data. The matrix contains data on 5,000
(downsampled) cells and the three gRNAs. The name of the
<code>grna_mat.mtx</code> file on the Parse website is
<code>CRISPR: Digital Gene Expression (DGE) Matrix</code>. Next,
<code>all_grnas.csv</code> is a <code>.csv</code> file with three
columns: “gene_id”, “gene_name”, and “genome”. (The column names are
misnomers.) The column “gene_id” gives the ID of each gRNA. Next, the
column “gene_name” gives the name of each gRNA. Finally, the column
“genome” indicates the “genome” with which each gRNA is associated. In
the case of <code>all_grnas.csv</code>, the “genome” column contains the
string “gRNA” across all rows. The name of the file of
<code>all_grnas.csv</code> on the Parse website is
<code>CRISPR: All Gene (CSV)</code>.</p>
<p>Next, <code>gene_mat.mtx</code> is a matrix market file containing
the gene expression data. The matrix contains data on 266 downsampled
genes and the same 5,000 (downsampled) cells as the gRNA matrix. The
name of <code>gene_mat.mtx</code> on the Parse website is
<code>Whole Transcriptome: Digital Gene Expression (DGE) Matrix</code>.
Finally, <code>all_genes.csv</code> is a <code>.csv</code> file
containing the columns “gene_id”, “gene_name”, and “genome”. The columns
“gene_id” and “gene_name” give the ID and human-readable name,
respectively, of each gene. The column “genome”, meanwhile, indicates
the reference genome to which the transcripts from each gene are mapped.
In <code>all_genes.csv</code> the “genome” column contains the string
“hg38” across all rows. Unfortunately, the file
<code>all_genes.csv</code> is not contained on the Parse website. We
(i.e., the maintainers of the <code>sceptredata</code> package) obtained
<code>all_genes.csv</code> directly from Parse.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" tabindex="-1"></a><span class="fu">library</span>(sessioninfo); <span class="fu">session_info</span>()</span></code></pre></div>
<pre><code>## ─ Session info ───────────────────────────────────────────────────────────────
##  setting  value
##  version  R version 4.3.3 (2024-02-29)
##  os       macOS Sonoma 14.3.1
##  system   aarch64, darwin20
##  ui       X11
##  language (EN)
##  collate  en_US.UTF-8
##  ctype    en_US.UTF-8
##  tz       America/New_York
##  date     2024-03-16
##  pandoc   3.1.1 @ /Applications/RStudio.app/Contents/Resources/app/quarto/bin/tools/ (via rmarkdown)
## 
## ─ Packages ───────────────────────────────────────────────────────────────────
##  package     * version date (UTC) lib source
##  bslib         0.6.1   2023-11-28 [2] CRAN (R 4.3.1)
##  cachem        1.0.8   2023-05-01 [2] CRAN (R 4.3.0)
##  cli           3.6.2   2023-12-11 [2] CRAN (R 4.3.1)
##  digest        0.6.34  2024-01-11 [2] CRAN (R 4.3.1)
##  evaluate      0.23    2023-11-01 [2] CRAN (R 4.3.1)
##  fastmap       1.1.1   2023-02-24 [2] CRAN (R 4.3.0)
##  htmltools     0.5.7   2023-11-03 [2] CRAN (R 4.3.1)
##  jquerylib     0.1.4   2021-04-26 [2] CRAN (R 4.3.0)
##  jsonlite      1.8.8   2023-12-04 [2] CRAN (R 4.3.1)
##  knitr         1.45    2023-10-30 [2] CRAN (R 4.3.1)
##  lattice       0.22-5  2023-10-24 [2] CRAN (R 4.3.3)
##  lifecycle     1.0.4   2023-11-07 [2] CRAN (R 4.3.1)
##  Matrix      * 1.6-5   2024-01-11 [2] CRAN (R 4.3.3)
##  R6            2.5.1   2021-08-19 [2] CRAN (R 4.3.0)
##  rlang         1.1.3   2024-01-10 [2] CRAN (R 4.3.1)
##  rmarkdown     2.26    2024-03-05 [2] CRAN (R 4.3.1)
##  rstudioapi    0.15.0  2023-07-07 [2] CRAN (R 4.3.0)
##  sass          0.4.8   2023-12-06 [2] CRAN (R 4.3.1)
##  sceptredata * 0.9.0   2024-03-17 [1] Bioconductor
##  sessioninfo * 1.2.2   2021-12-06 [2] CRAN (R 4.3.0)
##  xfun          0.42    2024-02-08 [2] CRAN (R 4.3.1)
##  yaml          2.3.8   2023-12-11 [2] CRAN (R 4.3.1)
## 
##  [1] /private/var/folders/7v/5sqjgh8j28lgf8qx3gbtq1h00000gp/T/RtmpNaqVEz/temp_libpath13ccc6abdb76e
##  [2] /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/library
## 
## ──────────────────────────────────────────────────────────────────────────────</code></pre>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
